File: app/src/main/java/com/lamnt/foodorder/utils/PreferenceUtil.java
package com.lamnt.foodorder.utils;

import android.content.Context;

public class PreferenceUtil {
    private static PreferenceUtil instance;
    private Context mContext;

    private PreferenceUtil(Context context) {
        this.mContext = context;
    }

    public static synchronized PreferenceUtil getInstance(Context context) {
        if (instance == null) {
            instance = new PreferenceUtil(context);
        }
        return instance;
    }
}

File: app/src/main/java/com/lamnt/foodorder/utils/PreferenceUtil.kt
package com.lamnt.foodorder.utils

import android.content.Context

class PreferenceUtil private constructor(private val mContext: Context) {

    companion object {
        private var instance: PreferenceUtil? = null

        @Synchronized
        @JvmStatic
        fun getInstance(context: Context): PreferenceUtil? {
            if (instance == null) {
                instance = PreferenceUtil(context)
            }
            return instance
        }
    }

}